pkg/controller/cyclenoderequest/transitioner/transitions.go

```diff
@@ -106,6 +109,19 @@ func (t *CycleNodeRequestTransitioner) transitionPending() (reconcile.Result, er
 
 	// get instances inside cloud provider node groups
 	nodeGroupInstances := nodeGroups.Instances()
+
+	forceCheckAsgConfigAnnotations := t.cycleNodeRequest.Annotations[ForceAsgConfigCheckAnnotation]
+	if forceCheckAsgConfigAnnotations == "true" {
+		for _, instance := range nodeGroupInstances {
+			outOfDate := instance.OutOfDate()
+			if outOfDate {
+				break
+			}
+		}
+		t.cycleNodeRequest.Status.Message = "Cycling skipped because all nodes in the nodegroups are up-to-date with a cloud provider ASG"
```

Comment:
does this get skipped when all instance are up to date? the break and completing the loop will both run this code?

Reply:
Ouch, right. This isn't how it's been tested in a lab. Let me push the original version

Reply:
@Jacobious52 I have introduce a bool skipped which is set to false once at least one out of date node is identified and then check it to decide if or not to transition a cnr to a Skipped state. Works fine is a lab now.

Reply:
Fixed by adding a skipped bool which is true by default and set to false if at least one node is not up to date. Tested in a lab by updating asg launch template, deleting a node, letting ASG recreate it so that I have 1 up to date and 2 outdated nodes.